### DISCUSSION TITLE
Cloudflare deploy

### DISCUSSION THREAD

**Discussion by dburrows** [wakujs/waku#400] [Q&A] [answered]
Love this project, trying to get it running on Cloudflare but the minimal instructions don't work for me with a clean project. May be doing something wrong but happy to write up a bit more documentation if someone can point me at a working Waku/Cloudflare project?

**Comment by dai-shi** [comment, marked as answer]
Can you elaborate a bit more please? It's unexpected to fail with a clean project. Yeah, a write-up would be super helpful.

**Comment by dburrows** [comment]
Generated a project from scratch and everything just worked perfectly through to deploy, will try and dig around and find out what was going wrong in the other one and if there's any gotchas will write it up.

Sorry for the noise, mark it up to Friday afternoon brain fog - really appreciate having a properly open-source (as opposed to company adjacent) next gen React framework that implements RSC etc.

**Reply by dai-shi** [reply]
Don't be sorry. It's actually great to hear someone tries cloudflare deploy.
I'm pretty sure, if you continue experimenting, you will find some issues.
Looking forward to getting those reports!

**Reply by jkhaui** [reply]
Hey this still looks to be a problem, should I open an issue?

Following the instructions I can get `create waku` working locally on windows and building artefacts for CloudFlare, but when running `wrangler deploy`, met with this error:

![image](https://github.com/dai-shi/waku/assets/36477474/2e223711-9bc2-4a1d-bd1a-0bb78c2e5f9e)

**Reply by dai-shi** [reply]
I just tried this
```bash
npm create waku@latest
cd ...
npm i
npm run build -- --with-cloudflare
npx wrangler dev
npx wrangler deploy
```

and deployed this: https://waku-project.muddy-unit-2d56.workers.dev

**Reply by jkhaui** [reply]
Hmm that's weird. I followed those steps exactly, only difference I used is pnpm (also using latest Node version of 22.3.0 and Windows).

Is this what my dist output should look like? Is there meant to be a `dist/serve.js` file here?
![image](https://github.com/dai-shi/waku/assets/36477474/eec88e21-db55-48e1-af14-df501eac3aff)

I'll try using npm and see if it makes a difference

**Reply by jkhaui** [reply]
Ok now it works, thanks!

I'm not sure which it was, but either downgrading node to 20.8.0 or switching to npm from pnpm has fixed it. I can also see the `dist/serve.js` file being created now

edit: turns out it's pnpm that for some reason leads to `serve.js` not being created in the build
